flask/app.py

```diff
@@ -790,25 +785,34 @@ def make_shell_context(self):
             rv.update(processor())
         return rv
 
-    def _reconfigure_for_run_debug(self, debug):
-        """The ``run`` commands will set the application's debug flag. Some
-        application configuration may already be calculated based on the
-        previous debug value. This method will recalculate affected values.
+    #: The environment value.  This is typically set from outside the
+    #: process by setting the `FLASK_ENV` environment variable and can be
+    #: used to quickly switch between different environments like
+    #: `production` and `development`.  If not set this defaults to
+    #: `production`.
+    env = ConfigAttribute('ENV')
 
-        Called by the :func:`flask.cli.run` command or :meth:`Flask.run`
-        method if the debug flag is set explicitly in the call.
+    def _get_debug(self):
+        return self.config['DEBUG']
+    def _set_debug(self, value):
+        self._set_debug_value(value)
 
-        :param debug: the new value of the debug flag
+    #: The debug flag.  Set this to ``True`` to enable debugging of the
+    #: application.  In debug mode the debugger will kick in when an unhandled
```

Comment:
"the debugger will kick in" is not accurate. It will only kick in if `FLASK_DEBUG=1` was passed so that the debug middleware is used. This is the source of a lot of confusion.

Reply:
@davidism that's why it's now documented that `DEBUG` must only be changed from the environment variable.

Reply:
Actually I missed some places, but the intention is that people flip this from the environment only.